src/pages/game/guess-bar/guessSlice.ts

```diff
@@ -1,23 +1,51 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 import { RootState } from '../../../app/store';
+import { guessStateType } from '../../../types/guessStateType';
+import { POLISH_COMMON_WORDS } from '../../../utils/constants';
 
-const initialState = {
-    text: '',
+const initialState: guessStateType = {
+    guessedWords: [],
+    guessHistory: [],
+    isAlreadyGuessed: false,
 };
 
 export const guessSlice = createSlice({
     name: 'guess',
     initialState,
     reducers: {
         setGuessText: (state, action: PayloadAction<string>) => {
-            state.text = action.payload;
+            const guessedWord = action.payload;
+            const updatedGuessWordsArray = [...state.guessedWords, guessedWord];
+
+            if (state.guessedWords.length === 0) {
+                return (state = {
+                    guessedWords: updatedGuessWordsArray,
+                    guessHistory: state.guessHistory,
+                    isAlreadyGuessed: false,
+                });
+            }
+
+            const isAlreadyGuessed =
+                state.guessedWords.includes(action.payload) ||
+                POLISH_COMMON_WORDS.includes(action.payload);
+
+            return (state = {
+                guessedWords: isAlreadyGuessed
+                    ? state.guessedWords
+                    : updatedGuessWordsArray,
+                guessHistory: state.guessHistory,
+                isAlreadyGuessed,
+            });
+        },
+        resetGame: (state) => {
```

Comment:
I would rename it and keep resetGame action for time when we would like to refetch article